### PR TITLE
Motorcycle Booking Backend

### PR DESCRIPTION
**Pull request by jonas-45**
This project is done by @ehmaddd, @Patfarmurs, @jonas-45, @Jawad-Nazari 
In this project, We implemented all the requirements as listed below:

Final Capstone: Motorcycle Booking App Backend
In this pull request, we aim to merge the following changes to the main branch of this repository:

- [ ] We created all tables pertaining to the project:
      - User. To store data on registered users (username)
      - Motorcycles. To store data on motorcycles(name, price, duration , etc.)
      - Reservations. To store data on reservations for motorcycles.

- [ ] We created various models to interact with the database including associations.
      - A motorcycle can have many reservations.
      - A reservation belongs to one motorcycle.

- [ ] We also created the API controllers with the respective methods to handle different requests.
      - [ ] Index. list all motorcycles and reservations.
      - [ ] Create. Allow the addition of a motorcycle, and reservations.
      - [ ] Show. Allow access to data of a specified motorcycle using their ID.
      - [ ] Destroy. Allow deletion of a motorcycle, with their reservations and reservations.

- [ ] Created routes
- [ ] Created the testing for the backend API.
- [ ] Created a documentation
       - To see the documentation, start the server and visit [link](http://localhost:3000/api-docs/index.html)
- [ ] Created and updated the readme file.

Link to the frontend repository: https://github.com/jonas-45/motorcycle-booking-react

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,187 @@
-# README
+<a name="readme-top"></a>
```
Comment (Meltrust) [inline README.md:1]: - [ ] Kindly, add a "tests" section on your readme detailing the correct way to run your tests.

In this case, the correct way would be `rake rswag`:

image

This command is idempotent, which means no matter how many times is run, the result will always be the same.  👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @jonas-45,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  API is working well✔️ 
-  Good readme ✔️ 




#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md (hunk 1 above)
Comment (Jawad-Nazari) [reply README.md:1]: Hi @Meltrust ,
Thanks for your time and for reviewing our code. We will update the readme file as you advised.

**Review by Meltrust — APPROVED** [review]
_note: This review outcome only applies to the backend repo.  Kindly, go to the [frontend ](https://github.com/jonas-45/motorcycle-booking-react/pull/13#pullrequestreview-1614171226)repo to see the review for that [project](https://github.com/jonas-45/motorcycle-booking-react/pull/13#pullrequestreview-1614171226)._

#### Hi @jonas-45,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Test section now exists on the readme✔️ 
- Nice job✔️ 




Cheers and Happy coding!👏👏👏


------

**Pull request merged**